src/api/v4/dataview/formula.js

```diff
@@ -0,0 +1,117 @@
+var _ = require('underscore');
+var Base = require('./base');
+var constants = require('../constants');
+var FormulaDataviewModel = require('../../../dataviews/formula-dataview-model');
+
+/**
+ * Formula dataview object
+ *
+ * @param {carto.source.Base} source - The source where the datavew will fetch the data.
+ * @param {string} column - The column name to get the data.
+ * @param {object} options
+ * @param {carto.operation} options.operation - The operation to apply to the data.
+ *
+ * @constructor
+ * @extends carto.dataview.Base
+ * @memberof carto.dataview
+ * @api
+ */
+function Formula (source, column, options) {
+  this._initialize(source, column, options);
+}
+
+Formula.prototype = Object.create(Base.prototype);
+
+/**
+ * Set the dataview operation
+ *
+ * @param  {carto.operation} operation
+ * @return {carto.dataview.Formula} this
+ * @api
+ */
+Formula.prototype.setOperation = function (operation) {
+  this._checkOperation(operation);
+  this._options.operation = operation;
+  if (this._internalModel) {
+    this._internalModel.set('operation', operation);
+  }
+  return this;
+};
+
+/**
+ * Return the current dataview operation
+ *
+ * @return {carto.operation} Current dataview operation
+ * @api
+ */
+Formula.prototype.getOperation = function () {
+  return this._options.operation;
+};
+
+/**
+ * Return the resulting data
+ *
+ * @return {FormulaData}
+ * @api
+ */
+Formula.prototype.getData = function () {
+  if (this._internalModel) {
+    /**
+     * @typedef {object} FormulaData
+     * @property {string} operation
+     * @property {number} result
+     * @property {number} nulls
+     * @property {string} type - Constant 'formula'
+     * @api
+     */
+    return {
+      operation: this._options.operation,
+      result: this._internalModel.get('data'),
+      nulls: this._internalModel.get('nulls'),
+      type: 'formula'
+    };
+  }
+  return null;
+};
+
+Formula.prototype._defaultOptions = function (options) {
+  options = options || {};
```

Comment:
[`_.defaults`](http://underscorejs.org/#defaults) might help here:

```
_.defaults(options, {
  operation: constants.operation.COUNT
});
```

We could actually do something like:

```
// base.js

Base.prototype.DEFAULTS = {};

Base.prototype._initialize = function () {
  ...
  this._options = _.defaults(options, this.DEFAULTS);
});

};

// formula.js

Formula.prototype.DEFAULTS = {
  operation: constants.operation.COUNT
}
```

Just an 💡

Reply:
Looks good to me